### PR TITLE
Feature: Ability to render at specific dateTime

### PR DESCRIPTION
**Pull request by JoshuaMoeckelmann**
### Description

Hey, 
I added the ability to change the time of the render. The initial value for the rendertime is the time that is set in the 'Picter render'.
Also i changed the gridbags to use a variable for their y-row, so its now easier to insert new rows, without needeing to shift all one down. Additionally I introduced a simpler variant of docChangeListener, that don't need as much codeduplication.

In the next days I will tackle the challange of a day/ night render, that automatically switches between them. :)

### How Has This Been Tested?

I ran different renders at varying times of day. They seem to corrispond to those, renderd via the photo renderer.

### Checklist

* [x] I have tested and built the changes locally and they work as expected
* [x] I have added relevant documentation or updated existing documentation
* [x] My changes generate no new warnings

### Screenshots (if applicable)

image

### Additional Context

None needed

### CODE REVIEW

**Comment by JoshuaMoeckelmann** [comment]
@shmuelzon may you have a look at the PR? :)

**Comment by shmuelzon** [comment]
Hey, first of all, thank you for contributing! I've added a few comments, please take a look and let me know what you think.

I was also wondering if there's any point in including a date in the UI, does it really make any difference? Maybe we can simplify it for the users by just displaying/modifying the time and, in the background, fix the date to 1/1/1970 or the project default. I somewhat doubt it makes much of a difference in the rendering, unless SH3D also knows the geographical location of the home and takes that into account.

**Comment by shmuelzon** [comment]
We'll also need to update the `doc/options.png` screenshot for the README file. If you're able, here's the demo project file I use for the documentation: [Demo.zip](https://github.com/user-attachments/files/16470658/Demo.zip)

**Comment by guimatheus92** [comment]
@JoshuaMoeckelmann have you made any improvements on this one so PR can get merged? I look forward for this change 😄

**Comment by JoshuaMoeckelmann** [comment]
> Hey, first of all, thank you for contributing! I've added a few comments, please take a look and let me know what you think.
> 
> I was also wondering if there's any point in including a date in the UI, does it really make any difference? Maybe we can simplify it for the users by just displaying/modifying the time and, in the background, fix the date to 1/1/1970 or the project default. I somewhat doubt it makes much of a difference in the rendering, unless SH3D also knows the geographical location of the home and takes that into account.

There is a difference based on the date and the location defined in the home object, the sun location changes for the rendering. The same is also visible in the normal photo editor... That's why I included it :) The location can be defined based on coordinates I guess by right-clicking on and defining it in the properties but don't quote me on that :D

Unfortunately I am unable to see your comments, maybe you didn't commit them, or just haven't reviewed it in github properly?

  


> @JoshuaMoeckelmann have you made any improvements on this one so PR can get merged? I look forward for this change 😄

@guimatheus92 i guess i have just put it off a bit because i was busy and couldn't see those coments @shmuelzon mentioned :D but if i see them I will fix em right away. I will also have a look at the automatic day night rendering this weekend :) 


FYI: With the newest home assistant update so 2024.8 there is a new graphical editor for the Picture elements Card, but tbh I still prefer this plugin, as it just makes the whole process extremely easy :)

**Review by shmuelzon — CHANGES_REQUESTED** [review]

**Comment by shmuelzon** [comment]
You're right, I didn't submit the review comments 🤦‍♂️

File: src/com/shmuelzon/HomeAssistantFloorPlan/Panel.java [hunk 1]
```diff
@@ -357,80 +373,89 @@ private void layoutComponents() {
         int labelAlignment = OperatingSystem.isMacOSX() ? JLabel.TRAILING : JLabel.LEADING;
         int standardGap = Math.round(2 * SwingTools.getResolutionScale());
         Insets insets = new Insets(0, standardGap, 0, standardGap);
+        int currentGridYIndex = 0;
 
-        /* First row (Detected lights caption) */
+        // Detected lights caption
```
[the remark this answers is not in the document]
Comment (JoshuaMoeckelmann) [reply src/com/shmuelzon/HomeAssistantFloorPlan/Panel.java:378]: Huh, you are the first person, that I've met, that prefers `/**/`. Is there a specific reason?

[the remark this answers is not in the document]
Comment (shmuelzon) [reply src/com/shmuelzon/HomeAssistantFloorPlan/Panel.java:378]: That was the convention at the first place I worked and it stuck with me 🤷‍♂️

**Comment by JoshuaMoeckelmann** [comment]
@shmuelzon I've implemented all of your suggestions :) could you have a look again?

**Review by shmuelzon — APPROVED** [review]

**Comment by shmuelzon** [comment]
Merged, thanks!

**Pull request merged**